google-api-client/src/main/java/com/google/api/client/googleapis/util/Utils.java

```diff
@@ -55,6 +67,68 @@ public static HttpTransport getDefaultTransport() {
     static final HttpTransport INSTANCE = new NetHttpTransport();
   }
 
-  private Utils() {
+  /**
+   * Returns the `cert_provider_command` field in context_aware_metadata.json file.
+   *
+   * @param contextAwareMetadata Input stream for ~/.secureConnect/context_aware_metadata.json file.
+   * @return `cert_provider_command` field
+   * @since 1.31
+   */
+  static List<String> extractCertificateProviderCommand(InputStream contextAwareMetadata)
+      throws IOException {
+    JsonParser parser = JsonFactoryInstanceHolder.INSTANCE.createJsonParser(contextAwareMetadata);
+    ContextAwareMetadataJson json = parser.parse(ContextAwareMetadataJson.class);
+    return json.getCommands();
   }
+
+  /**
+   * Returns if mTLS client certificate should be used. mTLS client certificate is used if and only
+   * if "GOOGLE_API_USE_CLIENT_CERTIFICATE" environment variable value is "true".
+   *
+   * @return If mTLS client certificate should be used
+   * @since 1.31
+   */
+  public static Boolean useMtlsClientCertificate() {
```

Comment:
This is an implementation detail of the mTLS detection workflow and shouldn't need to be public.